temporal-testing-junit4/src/main/java/io/temporal/testing/TestWorkflowRule.java

```diff
@@ -354,4 +354,27 @@ public History getWorkflowExecutionHistory(WorkflowExecution execution) {
   public Worker getWorker() {
     return testEnvironment.getWorkerFactory().getWorker(getTaskQueue());
   }
+
+  /**
+   * Returns the default test watcher impl that can be used is your tests. It simplifies the test watcher
+   * creation to just:
+   * <pre><code>
+   *   @Rule
+   *   public TestWatcher watchman = testWorkflowRule.getTestWatcher();
+   *   </code></pre>
+   *
+   * @return default test watcher impl
+   */
+  public TestWatcher getTestWatcher() {
+    return
+      new TestWatcher() {
+        @Override
+        protected void failed(Throwable e, Description description) {
+          if (getTestEnvironment() != null) {
+            System.err.println(getTestEnvironment().getDiagnostics());
+            getTestEnvironment().shutdown();
```

Comment:
Why do we need this? This code is located inside TestWorkflowRule class that takes care of shutting down the environment here, in shutdown call:
```
  @Override
  public Statement apply(Statement base, Description description) {
    taskQueue = init(description);
    Statement testWorkflowStatement =
        new Statement() {
          @Override
          public void evaluate() throws Throwable {
            start();
            base.evaluate();
            shutdown();
          }
        };
    return watchman.apply(globalTimeout.apply(testWorkflowStatement, description), description);
  }
```
So, the environment shutdown is taken care of already by the rule, why do we need also the watcher to handle it?

Reply:
This is a separate rule so from what i see needs to be there. On exception the test rule might not shut down.